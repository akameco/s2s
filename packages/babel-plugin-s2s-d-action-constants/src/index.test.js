// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const filename = path.resolve(process.cwd(), 'app', 'counter', 'actions.js')

beforeAll(() => {
  jest.resetModules()
  process.env.S2S_ACTION_CONSTANTS_PREFIX = 'app'
})

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: { filename },
  tests: [
    `export type Action = Increment`,
    `
    // @flow
    export type Action = Increment
    `,
    `export type Action = Increment | Decrement`,
    {
      title: 'imports',
      code: `
// @flow
import type { X } from './types';

export type Action = Increment
  `,
    },
    {
      title: 'snake_case',
      code: `export type Action = ThisIsAction;`,
    },
    {
      title: 'import type',
      code: `
import type { UserID } from './types.js'

export type FetchUser = {
  type: typeof FETCH_USER,
  id: UserID
};

export type Action = FetchUser;
      `,
    },
    {
      title: 'default',
      code: `
// @flow
import * as constants from './constants.js'

export type Action = Increment
    `,
    },
    {
      title: 'GenericTypeAnnotation',
      code: `
// @flow
import * as constants from './constants.js'

export type Action = string
    `,
    },
  ],
})
