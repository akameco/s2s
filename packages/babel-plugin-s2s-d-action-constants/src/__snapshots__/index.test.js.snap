// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1. s2s-d-action-constants: 1. s2s-d-action-constants 1`] = `
"
export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const INCREMENT: \\"INCREMENT\\" = \\"INCREMENT\\";
"
`;

exports[`default 2. s2s-d-action-constants: 2. s2s-d-action-constants 1`] = `
"
// @flow
export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const INCREMENT: \\"INCREMENT\\" = \\"INCREMENT\\";
"
`;

exports[`default 3. s2s-d-action-constants: 3. s2s-d-action-constants 1`] = `
"
export type Action = Increment | Decrement

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const INCREMENT: \\"INCREMENT\\" = \\"INCREMENT\\";
export const DECREMENT: \\"DECREMENT\\" = \\"DECREMENT\\";
"
`;

exports[`default GenericTypeAnnotation: GenericTypeAnnotation 1`] = `
"
// @flow
import * as constants from './constants.js'

export type Action = string

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
"
`;

exports[`default default: default 1`] = `
"
// @flow
import * as constants from './constants.js'

export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const INCREMENT: 'INCREMENT' = 'INCREMENT';
"
`;

exports[`default import type: import type 1`] = `
"
import type { UserID } from './types.js'

export type FetchUser = {
  type: typeof FETCH_USER,
  id: UserID
};

export type Action = FetchUser;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const FETCH_USER: 'FETCH_USER' = 'FETCH_USER';
"
`;

exports[`default imports: imports 1`] = `
"
// @flow
import type { X } from './types';

export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const INCREMENT: 'INCREMENT' = 'INCREMENT';
"
`;

exports[`default snake_case: snake_case 1`] = `
"
export type Action = ThisIsAction;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
export const THIS_IS_ACTION: \\"THIS_IS_ACTION\\" = \\"THIS_IS_ACTION\\";
"
`;
